src/community/infix-abstract.ts

```diff
@@ -0,0 +1,109 @@
+import { Notation } from "../notation";
+import type { DecimalSource } from "break_infinity.js";
+import Decimal from "break_infinity.js";
+import { fixMantissaOverflow } from "../utils";
+
+export abstract class AbstractInfixNotation extends Notation {
+  public name: string = "Abstract Infix";
+
+  public formatDecimal(value: Decimal, places: number): string {
+    return this.formatInfix(
+      value,
+      places
+    );
+  }
+
+  public format(value: DecimalSource, places: number = 0, placesUnder1000: number = 0): string {
+    if (typeof value === "number" && !Number.isFinite(value)) {
+      return this.infinite;
+    }
+
+    const decimal = Decimal.fromValue_noAlloc(value);
+
+    return decimal.sign() < 0
+      ? this.formatNegativeDecimal(
+        decimal.abs(),
+        places
+      )
+      : this.formatDecimal(
+        decimal,
+        places
+      );
+  }
+
+  protected groupDigits = 3;
+
+  protected canHandleZeroExponent = true;
```

Comment:
Codacy found an issue: [Declaration of protected instance field not allowed after declaration of public instance method. Instead, this should come after public instance fields.](https://app.codacy.com/gh/antimatter-dimensions/notations/pullRequest?prid=6218257)